### PR TITLE
Mbox mailbox driver support

### PR DESCRIPTION
**Pull request by TomasGalbickaNXP**
This PR adds new mbox driver for NXP mailbox IP.
Updates mbox related samples/tests for lpcxpresso55s69_cpu0 platform which is using new mbox_nxp_mailbox driver.

```
./scripts/twister --device-testing --device-serial /dev/ttyUSB0 --device-serial-baud 115200 -p lpcxpresso55s69_cpu0 --test tests/drivers/mbox/mbox_data/drivers.mbox_data --test samples/drivers/mbox/sample.drivers.mbox.real_hw --test samples/drivers/mbox_data/sample.drivers.mbox_data.real_hw --west-flash
INFO    - Using Ninja..
INFO    - Zephyr version: zephyr-v3.5.0-4976-gbb5574b505c9
INFO    - Using 'zephyr' toolchain.
INFO    - Building initial testsuite list...
INFO    - Writing JSON report /home/tom/zephyrproject/zephyr/twister-out/testplan.json

Device testing on:

| Platform             | ID   | Serial device   |
|----------------------|------|-----------------|
| lpcxpresso55s69_cpu0 |      | /dev/ttyUSB0    |

INFO    - JOBS: 20
INFO    - Adding tasks to the queue...
INFO    - Added initial list of jobs to queue
INFO    - Total complete:    3/   3  100%  skipped:    0, failed:    0, error:    0
INFO    - 3 test scenarios (3 test instances) selected, 0 configurations skipped (0 by static filter, 0 at runtime).
INFO    - 3 of 3 test configurations passed (100.00%), 0 failed, 0 errored, 0 skipped with 0 warnings in 19.60 seconds
INFO    - In total 6 test cases were executed, 0 skipped on 1 out of total 676 platforms (0.15%)
INFO    - 3 test configurations executed on platforms, 0 test configurations were only built.

Hardware distribution summary:

| Board                | ID   |   Counter |
|----------------------|------|-----------|
| lpcxpresso55s69_cpu0 |      |         3 |
```

### CODE REVIEW

File: samples/drivers/mbox/boards/lpcxpresso55s69_cpu0.overlay [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+		/* Attach MBOX driver to Mailbox Unit */
+        mbox:mailbox0@5008b000 {
+			compatible = "nxp,mbox-mailbox";
+			reg = <0x5008b000 0xEC>;
+			interrupts = <31 0>;
+			rx-channels = <4>;
+			#mbox-cells = <1>;
+			status = "okay";
+		};
+	};
```
Comment (carlocaione) [inline samples/drivers/mbox/boards/lpcxpresso55s69_cpu0.overlay:28]: the indentation looks odd (for every `.overlay` file in this commit)

File: drivers/mbox/mbox_nxp_mailbox.c [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+#define MAILBOX_ID_THIS_CPU  kMAILBOX_CM0Plus
+#define MAILBOX_ID_OTHER_CPU kMAILBOX_CM4
+#endif
+#endif
+
+#define GENIRQ_SHIFT     (28)
+#define GEN0_IRQ_TRIGGER (1U << (GENIRQ_SHIFT + 3U)) /*!< General interrupt 3. */
+#define GEN1_IRQ_TRIGGER (1U << (GENIRQ_SHIFT + 2U)) /*!< General interrupt 2. */
+#define GEN2_IRQ_TRIGGER (1U << (GENIRQ_SHIFT + 1U)) /*!< General interrupt 1. */
+#define GEN3_IRQ_TRIGGER (1U << (GENIRQ_SHIFT + 0U)) /*!< General interrupt 0. */
```
Comment (carlocaione) [inline drivers/mbox/mbox_nxp_mailbox.c:45]: Use `BIT()` macro

File: samples/drivers/mbox/boards/lpcxpresso55s69_cpu0.overlay (hunk 1 above)
Comment (TomasGalbickaNXP) [reply samples/drivers/mbox/boards/lpcxpresso55s69_cpu0.overlay]: Thanks, The sneaky whitespacess. Resolved.

File: drivers/mbox/mbox_nxp_mailbox.c (hunk 2 above)
Comment (TomasGalbickaNXP) [reply drivers/mbox/mbox_nxp_mailbox.c]: Cool hint! I used BIT and BIT_MASK.

**Review by carlocaione — DISMISSED** [review]

**Review by dbaluta — DISMISSED** [review]

**Review by iuliana-prodan — CHANGES_REQUESTED** [review]

File: drivers/mbox/mbox_nxp_mailbox.c [hunk 3]
```diff
@@ -0,0 +1,212 @@
+/*
+ * Copyright 2024 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Wrapper of NXP Mailbox driver for Zephyr's MBOX model.
+ */
+
+#include <zephyr/devicetree.h>
```
[the remark this answers is not in the document]
Comment (TomasGalbickaNXP) [reply drivers/mbox/mbox_nxp_mailbox.c:9]: `mbox_nxp_imx_mu` is for Messaging Unit (MU) IP block
`mbox_nxp_mailbox` is for Inter-CPU Mailbox IP block

The name `mbox_nxp_mailbox` follows defined naming - `mbox` driver, `nxp` vendor, `mailbox` the IP it uses same as for mu in `mbox_nxp_imx_mu`.
So this driver is compatible for any SOC which has Inter-CPU Mailbox IP block.

File: dts/bindings/mbox/nxp,mbox-mailbox.yaml [hunk 4]
```diff
@@ -0,0 +1,29 @@
+description: |
+  NXP Mailbox Unit as Zephyr MBOX
```
[the remark this answers is not in the document]
Comment (TomasGalbickaNXP) [reply dts/bindings/mbox/nxp,mbox-mailbox.yaml:2]: OK I will extend it a bit, thanks.

**Review by iuliana-prodan — DISMISSED** [review]

**Review by carlocaione — DISMISSED** [review]

**Review by dleach02 — DISMISSED** [review]

**Comment by TomasGalbickaNXP** [comment]
Few minor fixes, remove redundant code and add comment. https://github.com/zephyrproject-rtos/zephyr/compare/04b3809396844423abef20e9bd96ff5f3d070215..f9eceb8e34d91b89c10a3d2cac0b59d30c9cf992
And rebase to current main and fix merge conflict.

**Review by carlocaione — DISMISSED** [review]

**Review by iuliana-prodan — DISMISSED** [review]

**Review by dleach02 — APPROVED** [review]

**Comment by dleach02** [comment]
@iuliana-prodan @carlocaione, the HAL west.yml has been updated. Can we get your re-approvals.

**Review by iuliana-prodan — APPROVED** [review]

**Review by carlocaione — APPROVED** [review]

**Pull request merged**